File: frontend/src/types/reportcard.ts
// src/types/reportcard.ts
export interface AssessmentComponentScore {
  component_name: string;
  raw_marks_obtained: number;
  raw_maximum_marks: number;
  report_maximum_marks: number;
  converted_marks: number;
}

export interface SubjectAssessmentResult {
  assessment_name: string;
  components: AssessmentComponentScore[];
  total_obtained: number;
  total_maximum: number;
}

export interface SubjectReportCardDetail {
  subject_id: number;
  subject_name: string;
  assessments: SubjectAssessmentResult[];
  subject_total_obtained: number;
  subject_total_maximum: number;
  percentage: number;
  grade: string;
}

export interface ReportCardResponse {
  student_id: number;
  student_name: string;
  grade_id: number;
  section_id: number;
  academic_year_id: number;
  subjects: SubjectReportCardDetail[];
  grand_total_obtained: number;
  grand_total_maximum: number;
  overall_percentage: number;
  overall_grade: string;
  overall_result: 'Pass' | 'Fail';
  teacher_remarks?: string;
}

export interface ReportCardListResponse {
  total: number;
  items: ReportCardResponse[];
}

export interface ReportCardQueryParams {
  skip?: number;
  limit?: number;
  academic_year_id?: number;
  grade_id?: number;
  section_id?: number;
  student_id?: number;
}

File: backend-python/app/schemas/report_card.py
# app/schemas/report_card.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field

class AssessmentComponentScore(BaseModel):
    component_name: str
    raw_marks_obtained: float
    raw_maximum_marks: float
    report_maximum_marks: float
    converted_marks: float

class SubjectAssessmentResult(BaseModel):
    assessment_name: str
    components: List[AssessmentComponentScore]
    total_obtained: float
    total_maximum: float

class SubjectReportCardDetail(BaseModel):
    subject_id: int
    subject_name: str
    assessments: List[SubjectAssessmentResult]
    subject_total_obtained: float
    subject_total_maximum: float
    percentage: float
    grade: str

class ReportCardResponse(BaseModel):
    student_id: int
    student_name: str
    grade_id: int
    section_id: int
    academic_year_id: int
    subjects: List[SubjectReportCardDetail]
    grand_total_obtained: float
    grand_total_maximum: float
    overall_percentage: float
    overall_grade: str
    overall_result: str  # Pass / Fail
    teacher_remarks: Optional[str] = None

    class Config:
        from_attributes = True

class ReportCardRemarksUpdate(BaseModel):
    teacher_remarks: str = Field(..., description="Teacher remarks for the report card")

class ReportCardListResponse(BaseModel):
    total: int
    items: List[ReportCardResponse]
    """
SCHOLARIS ERP

Module:
Description:

TODO:
"""
